src/main/java/bio/terra/cda/app/service/QueryService.java

```diff
@@ -182,29 +195,32 @@ float elapsed() {
     return resultsCount;
   }
 
-  public List<JsonNode> runQuery(String query) {
-    QueryJobConfiguration queryConfig =
-        QueryJobConfiguration.newBuilder(query).setUseLegacySql(false).build();
+  private void logQuery(Job queryJob, List<JsonNode> jsonData) {
+    // Log usage data for this response.
+    final Map<Source, Integer> resultsCount = generateUsageData(jsonData);
+    var elapsed =
+        (queryJob.getStatistics().getEndTime() - queryJob.getStatistics().getStartTime()) / 1000.0F;
+    // This cast is safe because we've called getQueryResults() and that would throw if this job
+    // wasn't a query.
+    var queryConfig = (QueryJobConfiguration) queryJob.getConfiguration();
+    var logData = new QueryData(queryConfig.getQuery(), elapsed, resultsCount);
+    try {
+      logger.info(objectMapper.writeValueAsString(logData));
```

Comment:
Any chance this could get giant and become a problem?  Could this eventually be a security concern to log all of the data?

Reply:
It's only as big as a query is, I can add a ticket to check for queries larger than 100K or some other fairly large value. This is only logging the query string, the elapsed time, and the Map of data source and counts.

Reply:
I checked and BQ has a limit of 1024 characters for a query. Since we can only get here if BQ has already run the query, there's no need to worry about a lot of log data here. Note that the query response is not logged.